internal/cli/login.go

```diff
@@ -73,6 +90,41 @@ func (c *LoginCommand) Run(args []string) int {
 	return 0
 }
 
+func (c *LoginCommand) loginToken() (string, int) {
+	// First we decode the token to ensure it is valid and also to figure
+	// out if we have a login or invite token.
+	decodeResp, err := c.project.Client().DecodeToken(c.Ctx, &pb.DecodeTokenRequest{
+		Token: c.flagToken,
+	})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return "", 1
+	}
+	token := decodeResp.Token
+
+	// If we have a login token, then we're just done cause that can be stored directly.
+	if _, ok := token.Kind.(*pb.Token_Login_); ok {
+		return c.flagToken, 0
```

Comment:
Super yak shave-y question, but wouldn't `token` here and on line 188 make more semantic sense?

Reply:
`token` is the full `pb.Token` structure, not the string-encoded token 😄

Reply:
Ah ok, so is the `convertResp.Token` on line 125 a different type then than the `decodeResp.Token`?

Reply:
I'm not familiar with the larger OIDC picture yet 🙈 Tomorrow! hah

Reply:
Yeah haha. This is just the token system APIs so no need to know OIDC. `decodeResp` decodes a string token into a `*pb.Token` and Convert just returns the string token (converted from an invite token).